src/components/AddItemForm.tsx

```diff
@@ -1,7 +1,24 @@
-import React from 'react';
+import React, { useState } from 'react';
 
-function AddItemForm() {
+type Props = {
+  onAddTask: (taskTitle: string) => void;
+};
+
+function AddItemForm({ onAddTask }: Props) {
```

Comment:
Hey my friend, I have doubts regarding the need of the `domInputId` and `getElementById` here I believe we can turn the input into a controlled input and avoid that without losses
```jsx

function AddItemForm({ onAddTask }: Props) {
  const [inputText, setInputText] = useState('');

  const handleAdd = () => {
    if (inputText.trim() !== '') {
      onAddTask(inputText.trim());
      setInputText('');
    }
  };

  return (
    <form>
      <h2 className="label-wrapper">
        <label className="label__lg"> Lista de tarefas PodCodar </label>
      </h2>
      <input
        type="text"
        className="input input__lg"
        autoComplete="off"
        onChange={(e) => setInputText(e.target.value)}
        value={inputText}
      />
      <button
        type="button"
        className="btn btn__primary btn__lg"
        onClick={handleAdd}
      >
        Add Task
      </button>
    </form>
  );
}
```